### PR TITLE
Adding csv format parameters to network option in service create/update

### PR DESCRIPTION
**Pull request by abhi**
**- What I did**
Added capability to service create/update  to accept network option in csv format and add test case to verify the same.

**- How I did it**
Changed `--network` , `--network-add` options to accept csv format key/value by create a new network option type. 

**- How to verify it**

```
docker service create --name web --network name=docknet,alias=web1,alias=web2 nginx
docker service create --name web --network docknet nginx
docker service update web --network-add name=docknet,alias=web1
docker service update web --network-rm docknet
```

**- A picture of a cute animal (not mandatory but encouraged)**

polar-bear

### CODE REVIEW

**Comment by mavenugo** [comment]
ping @thaJeztah

**Comment by thaJeztah** [comment]
Thanks! Should we have the same format added to `docker container create/run` to keep parity?

**Comment by thaJeztah** [comment]
Also this relates to https://github.com/moby/moby/issues/31964 (although not all of the options are implemented yet)

**Comment by thaJeztah** [comment]
Hm, is this by design, or overlooked?

```bash
docker network create -d overlay docknet

docker service create \
  --name web \
  --network name=docknet,alias=web1,alias=web2 \
  --network name=docknet,alias=web3,alias=web4 \
  nginx:alpine
```

**Comment by thaJeztah** [comment]
Hm, looks like that's already possible in 17.05 (attach twice to the same network), gonna check 17.03/17.04

**Comment by abhinandanpb** [comment]
@thaJeztah : you beat me to it. Yes its possible in the current code as well. I can add a sanity check as well.
As Far as  #31964 I am in the process of extending the network options to have driver options as well. So the plan is to support network name , alias, driver specific option for the network.

**Comment by mavenugo** [comment]
@thaJeztah we should get this for `docker run` as well... but we can address it using another PR as these are fairly independent and also there is more work involved for `docker run` to support multiple networks.

And yes, this is related to #31964 the only 2 networks options that we support today for `docker service` are network `name` and `alias`. As we support other options, we can start to add to this CSV format.

File: opts/network.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+		key := strings.TrimSpace(strings.ToLower(parts[0]))
+		value := strings.TrimSpace(strings.ToLower(parts[1]))
+
+		switch key {
+		case networkOptName:
+			netAttach.Target = value
+		case networkOptAlias:
+			netAttach.Aliases = append(netAttach.Aliases, value)
+		default:
+			return fmt.Errorf("invalid field key %s", key)
```
Comment (thaJeztah) [inline opts/network.go]: I think we should make the error messages consistent with what we have for `--mount`, e.g. https://github.com/moby/moby/blob/49376cddab7d2ea28fd08fbbb75522656a72ed09/opts/mount.go#L129

File: opts/network.go [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+	for _, field := range fields {
+		parts := strings.SplitN(field, "=", 2)
+		//Support legacy non csv format
+		if len(parts) == 1 {
+			netAttach.Target = parts[0]
+			break
+		}
+
+		if len(parts) != 2 {
+			return fmt.Errorf("invalid field %s", field)
```
Comment (thaJeztah) [inline opts/network.go]: Same here for consistent error messages https://github.com/moby/moby/blob/49376cddab7d2ea28fd08fbbb75522656a72ed09/opts/mount.go#L82

File: opts/network.go [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+		switch key {
+		case networkOptName:
+			netAttach.Target = value
+		case networkOptAlias:
+			netAttach.Aliases = append(netAttach.Aliases, value)
+		default:
+			return fmt.Errorf("invalid field key %s", key)
+		}
+	}
+	n.networkAttachments = append(n.networkAttachments, netAttach)
```
Comment (thaJeztah) [inline opts/network.go]: Should we validate here if `netAttach.name` (or `ID`?) is set?

**Review by thaJeztah — CHANGES_REQUESTED** [review]
design LGTM - the validation for duplicate networks can be separate from this, but is probably something that needs to be fixed.

moving to code review (left some initial comments)

File: opts/network.go (hunk 2 above)
Comment (abhinandanpb) [inline opts/network.go]: @thaJeztah I was. following the error message https://github.com/moby/moby/blob/master/opts/port.go#L43

File: opts/network.go (hunk 1 above)
Comment (abhinandanpb) [inline opts/network.go]: same as above

File: opts/network.go (hunk 3 above)
Comment (abhinandanpb) [inline opts/network.go]: both work. So shouldn't be a problem

**Comment by abhinandanpb** [comment]
@thaJeztah thanks for taking a quick look. I will split this pr in docker/cli and here to continue the discussion further.

**Comment by abhinandanpb** [comment]
closing this in favor of #33130

**Pull request closed without merging**